.github/actions/meson/action.yml

```diff
@@ -0,0 +1,41 @@
+# A Github Action to build a repository with meson
+name: "Setup and run meson"
+description: "Setup and run meson"
+inputs:
+  builddir:
+    description: "The build directory"
+    default: "builddir"
+  srcdir:
+    description: "The source directory"
+    default: "."
+  meson_args:
+    description: "Arguments passed to meson setup"
+    default: ""
+  meson_skip_test:
+    description: "Set to a nonempty string to skip meson test"
+    default: ""
+  meson_test_args:
+    description: "Arguments passed to meson test"
+    default: ""
+  meson_precmd:
+    description: "Set to 'sudo' to run with sudo"
+    default: ""
+  ninja_args:
+    description: "Arguments passed to ninja"
+    default: ""
+  ninja_precmd:
+    description: "Set to 'sudo' to run with sudo"
+    default: ""
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        pushd ${{inputs.srcdir}}
```

Comment:
This is not needed, you can do `meson setup srcdir builddir` instead.

Eg. https://github.com/FFY00/mesonpy/blob/80563719cfafa11b059bb36fb56d2287f8d6e4e3/mesonpy.py#L436-L440

Reply:
good point, thanks. Fixed in 9bc2c60 but I'll squash that in before merging.

side-note: `meson setup builddir srcdir`, the two are swapped.